models/events.js

```diff
@@ -169,11 +171,62 @@ const kickoutPeer = async ({ eventId, peerId, reason }) => {
     throw error;
   }
 };
+/**
+ * Creates an events code document in the Firestore database with the given event code data.
+ * @async
+ * @function
+ * @param {object} eventCodeData - The data for the event code to be created.
+ * @throws {Error} If an error occurs while creating the event code document.
+ */
+
+const createEventCode = async (eventCodeData) => {
+  try {
+    const eventRef = eventModel.doc(eventCodeData.event_id);
+    const eventSnapshot = await eventRef.get();
+    const eventSnapshotData = eventSnapshot.data();
+    const createdAndUpdatedAt = admin.firestore.Timestamp.now();
+    const docRef = eventCodeModel.doc(eventCodeData.id);
+
+    await docRef.set({ ...eventCodeData, created_at: createdAndUpdatedAt, updated_at: createdAndUpdatedAt });
+
+    const docSnapshot = await docRef.get();
+    const data = docSnapshot.data();
+
+    const previouslyPresentEventCodes = eventSnapshotData?.event_codes?.byRole?.mavens && [
+      ...eventSnapshotData?.event_codes?.byRole?.mavens,
+    ];
+    if (!data) throw new Error();
+
+    if (previouslyPresentEventCodes?.length > 0) {
+      await eventRef.update({
+        event_codes: {
+          by_role: {
+            mavens: [...previouslyPresentEventCodes, data?.id],
```

Comment:
Is this `by_role` object used to extend the room code functionality to other users?

Reply:
Yes @SanketDhabarde